index.html

```diff
@@ -466,347 +466,387 @@ <h3>Ethereum EIP712 Signature 2021</h3>
-        ing"}],"Proof":[{"name":"verificationMethod","type":"string"},{"name":"created","type":"st
-        ring"},{"name":"proofPurpose","type":"string"},{"name":"type","type":"string"}],"Verifiabl
-        eCredential":[{"name":"@context","type":"string[]"},{"name":"type","type":"string[]"},{"na
-        me":"id","type":"string"},{"name":"issuer","type":"string"},{"name":"issuanceDate","type":
-        "string"},{"name":"credentialSubject","type":"CredentialSubject"},{"name":"credentialSchem
-        a","type":"CredentialSchema"},{"name":"proof","type":"Proof"}]}}
-      </pre>
+      <section>
+        <h3>Basic Document - Types Generation - Embedded Types</h3>
 
-      <p>The following is the resulting <code>proof</code> object:</p>
-      <pre class="example">
-        {
-          "type": "EthereumEip712Signature2021",
-          "created": "2021-07-09T19:47:41Z",
-          "proofPurpose": "assertionMethod",
-          "proofValue": "0x5fb8f18f21f54c2df8a2720d0afcee7dbbb18e4b7a22ce6e8183633d63b076d329122584db769cd78b6cd5a7094ede5ceaa43317907539187f1f0d8875f99e051b",
-          "verificationMethod": "did:example:aaaabbbb#issuerKey-1",
-          "eip712Domain": {
+        <p>
+          With the following <code>inputOptions</code> provided to the signature suite along with the <code>testBasicDocument</code> input document:
+        </p>
+
+        <pre class="example">
+          {
+            "date": "2021-08-30T13:28:02Z",
+            "verificationMethod": "did:pkh:eip155:1:0xAED7EA8035eEc47E657B34eF5D020c7005487443",
             "domain": {
-              "chainId": 4,
-              "name": "https://example.com",
-              "salt": "0x000000000000000000000000000000000000000000000000aaaabbbbccccdddd",
-              "version": "2"
-            },
-            "messageSchema": {
-              "CredentialSchema": [
-                {
-                  "name": "id",
-                  "type": "string"
-                },
-                {
-                  "name": "type",
-                  "type": "string"
-                }
-              ],
-              "CredentialSubject": [
-                {
-                  "name": "type",
-                  "type": "string"
-                },
+              "name": "Test"
+            }
+          }
+        </pre>
+
+        <p>The following is the resulting <code>proof</code> object:</p>
+
+        <pre class="example">
+          {
+            "created": "2021-08-30T13:28:02Z",
+            "proofPurpose": "assertionMethod",
+            "type": "EthereumEip712Signature2021",
+            "verificationMethod": "did:pkh:eip155:1:0xAED7EA8035eEc47E657B34eF5D020c7005487443",
+            "proofValue": "0x3d4518532cb589760742045bce8deb57dab474a3db500a7162b53ed43b034ef332e7e7f0c361140c366505e820ee4146f4fdd918e3dd27286f70df6435ab0b821c",
+            "eip712Domain": {
+              "domain": {
+                "name": "Test"
+              },
+              "messageSchema": {
```

Comment:
Its interesting because ethers.js throws an error if EIP712Domain is present, as it thinks it's not being used in the message due to different naming

Reply:
https://docs.ethers.io/v5/api/signer/#Signer-signTypedData

Reply:
I see ethers.js is constructing EIP712Domain and not allowing it to be passed explicitly:

https://github.com/ethers-io/ethers.js/blob/b1458989761c11bf626591706aa4ce98dae2d6a9/packages/hash/src.ts/typed-data.ts#L451-L469
https://github.com/ethers-io/ethers.js/blob/b1458989761c11bf626591706aa4ce98dae2d6a9/packages/hash/src.ts/typed-data.ts#L376-L389

It also "deduces" `primaryType`:

Reply:
This issue requires a major change to our implementation, did not know ahead of time this limitation of ethers.js. Unfortunately since we have deprioritized work on this as we chose to go with SIWE for our OCAP, I don't know how soon I would be able to get to this. 

On another note, since all the language and description is there, and the only thing missing is not including EIP712Domain, if you could push to my branch to get this merged, that would be pretty great too.

Reply:
Thanks for fixing the other things anyway. (empty domain object https://github.com/w3c-ccg/ethereum-eip712-signature-2021-spec/pull/26#discussion_r710068918, verification method DID URL https://github.com/w3c-ccg/ethereum-eip712-signature-2021-spec/pull/26#discussion_r723409980 https://github.com/w3c-ccg/ethereum-eip712-signature-2021-spec/pull/26#discussion_r704252673, and proof object https://github.com/w3c-ccg/ethereum-eip712-signature-2021-spec/pull/26#discussion_r710072926). Sorry it took me a while to respond. I didn't realize, or had forgotten, that these were fixed in 28bd5edecde8395242aea8ba64e9be25f59585d0.
Rather than push to this PR  branch, I've opened a new PR that extends this one: #43, as I feel that this would a more organized way to proceed.

Issue #41 is open to track the concern about Ethers.js compatibility vs. explicitly setting EIP712Domain (and primaryType).